### PR TITLE
Add tests for date and percent conditions

### PR DESCRIPTION
**Pull request by nkovshov**
I'm thinking to add #36 implementation, so I have added few tests for existing conditions.

### CODE REVIEW

File: gargoyle/conditions.py [hunk 1]
```diff
@@ -181,7 +182,7 @@ def clean(self, value):
         try:
             date = self.str_to_date(value)
         except ValueError as e:
-            raise ValidationError("Date must be a valid date in the format YYYY-MM-DD.\n(%s)" % e.message)
```
Comment (nkovshov) [inline gargoyle/conditions.py]: ValueError doesn't have `message` attribute in Python 3.5, this raised an AttributeError.

File: gargoyle/conditions.py [hunk 2]
```diff
@@ -181,7 +182,7 @@ def clean(self, value):
...
+            raise ValidationError("Date must be a valid date in the format YYYY-MM-DD.\n(%s)" % force_text(e))
```
Comment (adamchainz) [inline gargoyle/conditions.py]: I think it's better just to use `six.text_type`. `force_text` is only for things that could take lazy strings.

File: tests/testapp/test_conditions.py [hunk 3]
```diff
@@ -73,6 +75,40 @@ def test_clean_fail_too_many_numbers(self):
             condition.clean('1-2-3')
 
 
+class PercentTest(TestCase):
+    def test_clean(self):
```
Comment (adamchainz) [inline tests/testapp/test_conditions.py]: please split into multiple test functions. tests should be in individual functions so if they fail, we can see which pieces fail. here, if one fails, the remaining assertions aren't run. There's a pattern known as AAA, or Arrange Act Assert, you can read more about here: http://jamescooke.info/cleaner-unit-testing-with-the-arrange-act-assert-pattern.html

File: tests/testapp/test_conditions.py [hunk 4]
```diff
@@ -73,6 +75,40 @@ def test_clean_fail_too_many_numbers(self):
             condition.clean('1-2-3')
 
 
+class PercentTest(TestCase):
```
Comment (adamchainz) [inline tests/testapp/test_conditions.py]: naming convention dictates this be called `PercentTests` not `Test`

File: tests/testapp/test_conditions.py [hunk 5]
```diff
@@ -73,6 +75,40 @@ def test_clean_fail_too_many_numbers(self):
...
+            condition.clean('10-')
+        with pytest.raises(ValidationError):
+            condition.clean('-')
+        with pytest.raises(ValidationError):
+            condition.clean('')
+
+
+class DateTest(TestCase):
+    def test_abstract_date(self):
+        condition = AbstractDate()
```
Comment (adamchainz) [inline tests/testapp/test_conditions.py]: ditto the above. I think these would be cleaner as well with one test case per class, i.e. `AbstractDateTests` etc. then one test method per assertion

**Comment by nkovshov** [comment]
Adam, I don't sure what should I do to resume review process. 
I added changes, if it's ok, I also could squash commits.

**Comment by adamchainz** [comment]
Chill winston 😄  I've been busy going to Django Under The Hood ✨

**Comment by adamchainz** [comment]
Merged in 2714eb13e8e2837a5d09a68db1eaaacbc4df51bf, thanks!

**Pull request closed without merging**